Microsoft.Toolkit.Uwp.UI.Controls/Menu/Menu.Events.cs

```diff
@@ -0,0 +1,244 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Linq;
+using System.Text;
+using Windows.System;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Menu Control defines a menu of choices for users to invoke.
+    /// </summary>
+    public partial class Menu
+    {
+        private const string CtrlValue = "CTRL";
+        private const string ShiftValue = "SHIFT";
+
+        private static bool NavigateThroughMenuHeader(KeyEventArgs args, Menu menu, Orientation orientation)
+        {
+            if (orientation == Orientation.Horizontal)
+            {
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    menu.SelectedHeaderItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Left)
+                {
+                    MoveFocusBackwardAndGetPrevious(menu);
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    MoveFocusForwardAndGetNext(menu);
+                    return true;
+                }
+            }
+            else
+            {
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    menu.SelectedHeaderItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Up)
+                {
+                    MoveFocusBackwardAndGetPrevious(menu);
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    MoveFocusForwardAndGetNext(menu);
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool NavigateThroughMenuFlyoutItems(KeyEventArgs args, Menu menu, Orientation orientation)
+        {
```

Comment:
MenuFlyoutSubItem is overriding my code, will check if I can override MenuFlyoutSubItem navigation though code